csharp/src/Ice/Reference.cs

```diff
@@ -740,13 +741,9 @@ static void StartQueryOption(StringBuilder sb, ref bool firstOption)
             {
                 cancel.ThrowIfCancellationRequested();
                 InvocationInterceptor? interceptor = null;
-                if (i < proxyInterceptors.Count)
-                {
-                    interceptor = proxyInterceptors[i];
-                }
-                else if (i - proxyInterceptors.Count < communicatorInterceptors.Count)
+                if (i < invocationInterceptors.Count)
                 {
-                    interceptor = communicatorInterceptors[i - proxyInterceptors.Count];
+                    interceptor = invocationInterceptors[i];
                 }
```

Comment:
Now that we have a single list we can simplify this a bit further

```
if (i < invocationInterceptors.Count)
{
    // Call the next interceptor in the chain
    InvocationInterceptor interceptor = invocationInterceptors[i];
    return await interceptor(
```

Reply:
Fixed.